library/src/main/java/io/ona/kujaku/manager/DrawingManager.java

```diff
@@ -0,0 +1,652 @@
+        return new KujakuCircleOptions()
+                .withCircleRadius(10.0f)
+                .withCircleColor("black")
+                .withMiddleCircle(true)
+                .withDraggable(false);
+    }
+
+    public static KujakuCircleOptions getKujakuCircleDraggableOptions() {
+        return new KujakuCircleOptions()
+                .withCircleRadius(20.0f)
+                .withCircleColor("red")
+                .withMiddleCircle(false)
+                .withDraggable(true);
+    }
+
+    /**
+     * Drawing enabled/disabled
+     *
+     * @param drawing
+     */
+    private void setDrawing(boolean drawing) {
+        this.drawingEnabled = drawing;
+    }
+
+    /**
+     * Set currentCircle when circle is clicked or longClicked
+     *
+     * @param circle
+     */
+    private void setCurrentCircle(@Nullable Circle circle) {
+        this.currentKujakuCircle = getKujakuCircle(circle);
+    }
+
+    /**
+     * Set currentCircle when circle is clicked or longClicked
+     *
+     * @param circle
+     */
+    private void setCurrentKujakuCircle(@Nullable KujakuCircle circle) {
+        this.currentKujakuCircle = circle;
+    }
+
+    /**
+     * Start Drawing. A KujakuLayer can be passed to init the drawing.
+     * @param kujakuLayer
+     * @return
+     */
+    public boolean startDrawingKujakuLayer(@Nullable KujakuLayer kujakuLayer) {
+        if (kujakuLayer == null) {
+            this.startDrawingPoints(new ArrayList<>());
+            return true;
+        }
+
+        Geometry geometry = kujakuLayer.getFeatureCollection().features().get(0).geometry();
+        if (geometry instanceof Polygon) {
+            kujakuLayer.removeLayerOnMap(mapboxMap);
+            Polygon polygon = (Polygon) geometry;
+            List<Point> points = polygon.coordinates().get(0);
+            this.startDrawingPoints(points);
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Start Drawing. A list of point can be passed to init the drawing.
+     * Start drawing on the map / or editing an already existing kujakuLayer
+     *
+     * @param points
+     */
+    public void startDrawingPoints(@Nullable List<Point> points) {
+        setDrawing(true);
+        setCurrentCircle(null);
+
+        if (points != null && points.size() > 0) {
+            for (Point p : points) {
+                this.create(getKujakuCircleOptions().withLatLng(new LatLng(p.latitude(), p.longitude())), false);
+            }
+
+            this.refresh(true);
+        }
+    }
+
+    /**
+     * Stop drawing and create a new Kujaku layer
+     *
+     * @return
+     */
+    public boolean stopDrawingAndDisplayLayer() {
+        Polygon polygon = this.stopDrawing();
+
+        Feature feature = Feature.fromGeometry(polygon);
+        FeatureCollection collection = FeatureCollection.fromFeature(feature);
+        FillBoundaryLayer layer = new FillBoundaryLayer.Builder(collection)
```

Comment:
So can we reuse the same `FillBoundaryLayer`. Seems that this creates multiple `FillBoundaryLayers`

Reply:
Yes sure, we can create one if it is a new draw, and update the existing one if it is an edition

Reply:
done